### PR TITLE
docs: Update doc about CSV upload

### PR DESCRIPTION
**Pull request by seiyab**
### SUMMARY

Docs about CSV upload is stale. I update it to match current UI.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

NOTE: not full difference.
Before:
image

After:
image

### TESTING INSTRUCTIONS

N/A. Just updating doc.

### ADDITIONAL INFORMATION

- [x] Has associated issue: #14143
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: docs/docs/using-superset/exploring-data.mdx [hunk 1]
```diff
@@ -27,33 +27,34 @@ following information about each flight is given:
 You may need to enable the functionality to upload a CSV or Excel file to your database. The following section
 explains how to enable this functionality for the examples database.
 
-In the top menu, select **Data ‣ Databases**. Find the **examples** database in the list and
+In the top menu, select **Settings ‣ Data ‣ Databases Connections**. Find the **examples** database in the list and
```
Comment (rusackas) [inline docs/docs/using-superset/exploring-data.mdx:30]:
```suggestion
In the top menu, select **Settings ‣ Data ‣ Database Connections**. Find the **examples** database in the list and
```

File: docs/docs/using-superset/exploring-data.mdx [hunk 2]
```diff
@@ -27,33 +27,34 @@ following information about each flight is given:
...
 
-<img src={useBaseUrl("/img/tutorial/add-data-upload.png" )} />
+<img src={useBaseUrl("/img/tutorial/allow-file-uploads.png" )} />
 
 ### Loading CSV Data
 
 Download the CSV dataset to your computer from
 [GitHub](https://raw.githubusercontent.com/apache-superset/examples-data/master/tutorial_flights.csv).
-In the Superset menu, select **Data ‣ Upload a CSV**.
+In the top menu, select **Settings ‣ Data ‣ Databases Connections**. Then, **Upload file to database ‣ Upload CSV**.
```
Comment (rusackas) [inline docs/docs/using-superset/exploring-data.mdx:44]:
```suggestion
In the top menu, select **Settings ‣ Data ‣ Database Connections**. Then, **Upload file to database ‣ Upload CSV**.
```

**Review by rusackas — APPROVED** [review]
LGTM! Thank you for the help here! 

And congrats on the first commit. Feel free to add your org to the [In the Wild Page](https://github.com/apache/superset/blob/master/RESOURCES/INTHEWILD.md), too!

**Comment by seiyab** [comment]
Thank you for your review.
Our organization has not determined to use Superset yet. I will add it when I will contribute Superset again after our org stating use it. Thanks.

**Pull request merged**